From 6202c6032f17438a2facb23f02e330b9d0566d9d Mon Sep 17 00:00:00 2001
From: Thomas Koutcher <thomas.koutcher@online.fr>
Date: Sun, 24 Nov 2019 13:06:08 +0100
Subject: [PATCH] Fix segmentation fault

As reported by Maxim Zhukov, the commit 9ca7710 ("Fix memory leak in
main view (#931)", 01-06-2019) causes a SIGSEGV when holding the key
'Stage/Unstage current line' in stage view.

Add a check to avoid null pointer dereference.

Fixes #971
---
 src/graph-v2.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/graph-v2.c b/src/graph-v2.c
index 46504a85..69ac1587 100644
--- src/graph-v2.c
+++ src/graph-v2.c
@@ -257,6 +257,7 @@ done_graph(struct graph *graph_ref)
 {
 	struct graph_v2 *graph = graph_ref->private;

-	htab_delete(graph->colors.id_map);
+	if (graph->colors.id_map)
+		htab_delete(graph->colors.id_map);

 	free(graph);
